x/foundation/keeper/genesis.go

```diff
@@ -114,6 +114,11 @@ func (k Keeper) InitGenesis(ctx sdk.Context, sk foundation.StakingKeeper, data *
 
 	k.SetPool(ctx, data.Pool)
 
+	if data.OneTimeMintMaxCount != 1 {
+		return errors.New("OneTimeMintMaxCount must be 1")
+	}
```

Comment:
You may remove the validation logic thanks to `ValidateGenesis()`.

Reply:
I removed that logic.